src/Slider/Slider.js

```diff
@@ -0,0 +1,150 @@
+import { Box, Slider as MuiSlider, Typography } from "@mui/material";
+import PropTypes from "prop-types";
+import React from "react";
+
+/**
+ * Discrete slider component
+ */
+export default function Slider({
+  displayCurrentValue = "auto",
+  max = 10,
+  min = 1,
+  onChange = () => {},
+  onChangeCommitted = () => {},
+  orientation = "horizontal",
+  predefValues = [],
+  labels = true,
+  labelPosition = "bottom",
+  step = 1,
+  title,
+  value
+}) {
+  // get ticks labels if they are not pre defined
+  const range = (start, stop, step) =>
+    Array.from(
+      { length: (stop - start) / step + 1 },
+      (_, i) => start + i * step
+    );
+  // show/hide labels
+  let marks = [];
+  let stepSize = step;
+  if (labels) {
+    if (!predefValues.length) {
+      range(min, max, stepSize).map(item => {
+        const thisMark = { label: String(item), value: item };
+        marks.push(thisMark);
+      });
+    } else {
+      marks = predefValues;
+      stepSize = null;
+    }
+  } else {
+    marks = false;
+  }
+  // return components
+  return (
+    <Box>
+      <Typography>{title}</Typography>
+      {labelPosition === "top" ? (
```

Comment:
Would be cleaner to remove this ternary here, just have one instance of MuiSlider and then handle the sx styling as a ternary based on label position. Otherwise you are just duplicating code and could forget to add prop to both instances if any changes are made in the future.